### PR TITLE
Jordan/bueno welcome

### PR DESCRIPTION
**Pull request by jbibla**
**Description:**
- moved router-view into app so it could be used as a modal overlay and turned session frame into a session modal
- transformed card sign in, into nice welcome screen with session options to address user feedback and unwelcoming onboarding experience
- removed SessionWelcome screen 
- fixed a bunch of session related inconsistencies and styling issues
- select network modal uses same styles as network page 

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

**Comment by jbibla** [comment]
Screen Shot 2020-04-01 at 9 12 50 PM
Screen Shot 2020-04-01 at 9 12 58 PM
Screen Shot 2020-04-01 at 9 13 20 PM
Screen Shot 2020-04-01 at 9 13 28 PM

left to do:
- fix tests
- remove TmSessionWelcome
- check extension

File: src/components/common/CardSignInRequired.vue [hunk 1]
```diff
@@ -1,25 +1,107 @@
...
-    </div>
-    <div slot="subtitle">
-      This page shows data for a specific address. To access the page, please
-      <a @click="$router.push(`/welcome`)">sign&nbsp;in</a> or
-      <a @click="$router.push(`/welcome`)">sign&nbsp;up</a>.
+  <div class="card-sign-in">
+    <h2>Welcome to Lunie</h2>
+    <h3>How would you like to get started?</h3>
+
+    <div class="session-list">
```
Comment (faboweb) [inline src/components/common/CardSignInRequired.vue:6]: this now creates a lot of dupe code

File: src/components/common/TmSessionExplore.vue [hunk 2]
```diff
@@ -212,7 +212,7 @@ export default {
       if (addressType === "explore") return `Explore Mode`
       if (addressType === "ledger") return `Ledger Nano S`
       if (addressType === "extension") return `Lunie Browser Extension`
-      if (addressType === "local") return `Mobile App`
+      if (addressType === "local") return `Local`
```
Comment (faboweb) [inline src/components/common/TmSessionExplore.vue:215]: Nice

File: src/components/common/CardSignInRequired.vue (hunk 1 above)
Comment (jbibla) [reply src/components/common/CardSignInRequired.vue:6]: i can delete the other components where this lived originally `TmSessionExisting`

...and a refactor is in order, but I'm trying to deliver the value of a friendlier home screen for now.

Comment (faboweb) [reply src/components/common/CardSignInRequired.vue:6]: if you want to defer this, please create a refactoring issue and let's do it soon.

Comment (jbibla) [reply src/components/common/CardSignInRequired.vue:6]: > this now creates a lot of dupe code

no it doesn't, I moved this from `TmSessionExisting` and will delete it!

**Comment by faboweb** [comment]
There is a weird delay when clicking on a sidemenu link to switch the pages. (slow internet though)

**Comment by faboweb** [comment]
Also when clicking on one of the buttons

**Comment by faboweb** [comment]
issue with network selector on Windows
image

**Comment by faboweb** [comment]
Select Network in account creation comes before the security block. Is that intended?

**Comment by jbibla** [comment]
> issue with network selector on Windows

thanks! i will solve! 

> Select Network in account creation comes before the security block

ah interesting. no this wasn't intended. in secure mode we should go straight to the security block (which I hope to revamp in a new PR one day).

**Comment by jbibla** [comment]
fixing mario's bug and mobile + extension still... please don't merge yet.

File: src/components/common/TmSessionExplore.vue [hunk 3]
```diff
@@ -96,6 +96,11 @@ import TmFormMsg from "common/TmFormMsg"
 import bech32 from "bech32"
 import { formatAddress } from "src/filters"
 
+const isPolkadotAddress = address => {
```
Comment (faboweb) [inline src/components/common/TmSessionExplore.vue:99]: lol, all the logic in this file should move out. instead address format errors should be bubbled up from the common function in the store (connection.js > getNetworkByAccount)

Comment (jbibla) [reply src/components/common/TmSessionExplore.vue:99]: @Bitcoinera why are you doing this on this PR? was it causing an issue here?

Comment (Bitcoinera) [reply src/components/common/TmSessionExplore.vue:85]: I saw the bug on the Kanban and I am wondering just like you why this `this.isPolkadotAddress` is here 🤷‍♀ 

Very odd

Comment (Bitcoinera) [reply src/components/common/TmSessionExplore.vue:85]: I really like this PR's name btw 😸

File: package.json [hunk 4]
```diff
@@ -19,7 +19,7 @@
     "test:coverage": "live-server ./coverage/lcov-report",
     "test:coverage-branch": "npm run test:unit -- --coverage && git diff origin/develop | diff-test-coverage -c ./coverage/lcov.info -t lcov --",
     "test:e2e:browserstack": "nightwatch -c tests/e2e/browserstack.conf.js --suiteRetries 3",
-    "test:e2e:local": "nightwatch --config ./tests/e2e/local-chrome.conf.js",
+    "test:e2e:local": "npx kill-port 9515 && nightwatch --config ./tests/e2e/local-chrome.conf.js",
```
Comment (faboweb) [inline package.json:22]:
```suggestion
    "test:e2e:local": "nightwatch --config ./tests/e2e/local-chrome.conf.js",
```
this doesn't work properly on Windows

File: src/components/network/NetworkList.vue [hunk 5]
```diff
@@ -1,15 +1,18 @@
...
-    >
-      <NetworkItem :networkitem="network" :disabled="disabled" />
-    </li>
-  </ul>
+  <section>
+    <h3>{{ sectionTitle }}</h3>
+    <ul class="network-list">
+      <li
+        v-for="network in networks"
+        :key="network.chain_id"
```
Comment (faboweb) [inline src/components/network/NetworkList.vue:7]:
```suggestion
        :key="network.id"
```

File: src/components/staking/PageValidator.vue [hunk 6]
```diff
@@ -595,7 +595,7 @@ span {
   line-height: normal;
 }
 
-.bech32-address {
+.copyable-address {
```
Comment (faboweb) [inline src/components/staking/PageValidator.vue:598]: why is this defined multiple times?

File: src/styles/session.css [hunk 7]
```diff
@@ -171,9 +139,9 @@
     display: none;
   }
 
-  .session-close-mobile {
+  /* .session-close-mobile {
     display: none;
-  }
+  } */
```
Comment (faboweb) [inline src/styles/session.css:144]: remove?

File: tests/e2e/helpers.js [hunk 8]
```diff
@@ -252,8 +252,8 @@ async function getAccountBalance(browser) {
         browser.globals.denom = total[1]
         browser.globals.totalAtoms = total[0]
       })
-      /*await browser.getText(".total-atoms h2", result => {
-        browser.globals.totalAtoms = result.value.replace(",", "")
+      /*await browser.getText(".total-atoms h2", result => {
```
Comment (faboweb) [inline tests/e2e/helpers.js:255]: remove?

File: tests/unit/specs/components/common/TmSelectNetwork.spec.js [hunk 9]
```diff
@@ -52,51 +52,9 @@ describe(`TmSelectNetwork`, () => {
...
-        networks
-      )
-    })
   })
 
   it(`has the expected html structure`, async () => {
     expect(wrapper.element).toMatchSnapshot()
   })
-
-  it(`returns the networks sorted by the following criteria: current first, then default then mainnets, then testnets`, () => {
```
Comment (faboweb) [inline tests/unit/specs/components/common/TmSelectNetwork.spec.js:68]: did u change this?

File: tests/unit/specs/components/network/NetworkList.spec.js [hunk 10]
```diff
@@ -52,19 +53,36 @@ describe(`NetworkList`, () => {
...
 
-  it("sets new network when clicking list item", () => {
-    wrapper.find(".select-network-item:not(.selected)").trigger("click")
-    expect(wrapper.vm.$store.dispatch).toHaveBeenCalledWith(`setNetwork`, {
-      id: "gaia-testnet",
-      chain_id: "gaia-123",
-      logo_url: "cosmos-logo.png",
-      testnet: true,
-      title: "Cosmos Hub Test"
+  it(`returns an empty string`, () => {
```
Comment (faboweb) [inline tests/unit/specs/components/network/NetworkList.spec.js:56]: this test doesn't say anything. What should it actually achieve?

File: tests/unit/specs/components/network/NetworkList.spec.js [hunk 11]
```diff
@@ -52,19 +53,36 @@ describe(`NetworkList`, () => {
...
+    expect(wrapper.vm.whichFlow).toBe("")
+  })
+
+  it(`returns the create route as a string`, () => {
```
Comment (faboweb) [inline tests/unit/specs/components/network/NetworkList.spec.js:60]: when?

File: tests/unit/specs/components/network/NetworkList.spec.js [hunk 12]
```diff
@@ -52,19 +53,36 @@ describe(`NetworkList`, () => {
...
+    wrapper.setData({ $route: { name: `select-network-create` } })
+    expect(wrapper.vm.whichFlow).toBe("/create")
+  })
+
+  it(`returns the recover route as a string`, () => {
```
Comment (faboweb) [inline tests/unit/specs/components/network/NetworkList.spec.js:65]: same

File: package.json (hunk 4 above)
Comment (jbibla) [reply package.json]: sorry, didn't mean to leave this in. was an issue every time I ran e2e.

File: src/components/staking/PageValidator.vue (hunk 6 above)
Comment (jbibla) [reply src/components/staking/PageValidator.vue:598]: /shrug — probably needs to be refactored

File: src/styles/session.css (hunk 7 above)
Comment (jbibla) [reply src/styles/session.css:144]: yes, thanks.

File: tests/e2e/helpers.js (hunk 8 above)
Comment (jbibla) [reply tests/e2e/helpers.js:255]: yes, thanks.

File: tests/unit/specs/components/common/TmSelectNetwork.spec.js (hunk 9 above)
Comment (jbibla) [reply tests/unit/specs/components/common/TmSelectNetwork.spec.js:68]: [yes](https://github.com/luniehq/lunie/pull/3830/files/396b1f2dd91ef2d71929503ca49be7fb2e241b87#diff-7054cab35095f712da18c2b3232f118bL66). i thought it was overcomplicated and unnecessary. network list always appears the same way now.

File: src/ActionModal/components/ActionModal.vue [hunk 13]
```diff
@@ -639,7 +639,8 @@ export default {
     goToSession() {
       this.close()
 
-      this.$router.push(`/portfolio`)
+      this.$store.dispatch(`signOut`, this.network)
```
Comment (faboweb) [inline src/ActionModal/components/ActionModal.vue:642]: in which case is this needed?

Comment (jbibla) [reply src/ActionModal/components/ActionModal.vue:642]: this is when a user is in explore mode. if you open the action modal in explore mode you see the sign in button. so you have to sign out and then you can see the options for signing in. we used to send the user to the welcome screen but this doesn't exist anymore.

**Pull request merged**